dc/s2n-quic-dc/src/stream/client/tokio.rs

```diff
@@ -65,13 +65,21 @@ where
     // Make sure TCP_NODELAY is set
     let _ = socket.set_nodelay(true);
 
+    // if the acceptor_ip isn't known, then ask the socket to resolve it for us
+    let peer_addr = if acceptor_addr.ip().is_unspecified() {
+        socket.peer_addr()?
+    } else {
+        acceptor_addr
+    }
+    .into();
```

Comment:
IIRC, we were seeing a similar ~bug in the handshake metrics -- maybe we need code like this somewhere in s2n-quic too?

Reply:
Yeah I can look into that